packages/connection-manager/src/store/AbstractConnectionStore.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  IConnection,
+  IConnectionParty
+} from '@sphereon/ssi-sdk-core'
+
+export abstract class AbstractConnectionStore {
+  abstract getParty(partyId: string): Promise<IConnectionParty>
+  abstract getParties(): Promise<Array<IConnectionParty>>
+  abstract addParty(name: string): Promise<IConnectionParty>
+  abstract updateParty(party: IConnectionParty): Promise<IConnectionParty>
+  abstract removeParty(partyId: string): Promise<void>
+  abstract getConnection(connectionId: string): Promise<IConnection>
+  abstract getConnections(partyId: string): Promise<Array<IConnection>>
+  abstract addConnection(partyId: string, connection: Omit<IConnection, 'id' | 'createdAt' | 'lastUpdatedAt' | 'config.id' | 'identifier.id'>): Promise<IConnection>
```

Comment:
Would be nicer to create an interface for the addConnection instead of ommiting properties here from a persisted connection

Reply:
refactored to interface